almdrlib/environment.py

```diff
@@ -64,32 +67,23 @@ class AlmdrlibSourceNotEnabledError(AlEnvException):
 
 
 class AlEnv:
+
+    dynamodb = None
+    ssm = None
+    table = None
+    table_name = None
+
     def __init__(self, application_name, client=None, source=("dynamodb",)):
         self.application_name = application_name
         self.client = client
         self.source = source if type(source) in [tuple, list] else (source,)
-        self.region = AlEnv._get_region()
-        self.stack_name = AlEnv._get_stack_name()
-        self.table_name = AlEnv._table_name(self.region, self.stack_name)
-        try:
-            self.dynamodb = boto3.resource('dynamodb')
-            self.ssm = boto3.client('ssm')
-        except (botocore.exceptions.NoRegionError, botocore.exceptions.NoCredentialsError) as e:
-            raise AlEnvAwsConfigurationException(f'Please validate your AWS configuration') from e
-        if "dynamodb" in source:
-            try:
-                self.table = self.dynamodb.Table(self.table_name)
-                self._table_date_time = self.table.creation_date_time
-            except botocore.exceptions.ClientError as e:
-                if e.response['Error']['Code'] == 'ResourceNotFoundException':
-                    raise AlEnvConfigurationTableUnavailableException(self.table_name)
-                else:
-                    raise AlEnvException() from e
+
+        self._setup_aws_resources(source)
 
     def get(self, key, default=None, format='decoded', type=None):
```

Comment:
Perhaps this should be get_from_dynamodb, and the other should be get_from_ssm - as it's not clear just looking at https://github.com/alertlogic/alertlogic-sdk-python/blob/6f49dbf9d482fc4fb88ec2bd23c4139be1f52bfb/almdrlib/config.py#L119-L128 why one uses env.get and one uses env.get_parameter